src/clients/cli.rs

```diff
@@ -25,9 +26,42 @@ pub struct Cli {
     /// directly fetch the logs of a container.
     container_startup_timestamps: RwLock<HashMap<String, Instant>>,
     created_networks: RwLock<Vec<String>>,
+    command: OsString,
+}
+
+impl Default for Cli {
+    fn default() -> Self {
+        Self::docker()
+    }
 }
 
 impl Cli {
+    /// Create a new client, providing a custom `docker` command.
+    pub fn new<S>(command: S) -> Self
+    where
+        S: Into<OsString>,
```

Comment:
What is the need for `OsString`? Why not just a normal `String`?

Reply:
The `new` function of `Command` is defined as:

```rust
    pub fn new<S: AsRef<OsStr>>(program: S) -> Command {
        Command { inner: imp::Command::new(program.as_ref()) }
    }
```

I used an `OsString`, rather than `OsStr`, in order to clone it and be able to simply pass it into the `new` function, as it already is an `OsString`. This way testcontainers doesn't make an decision on the data type, as it only passes through the value from the user to the `Command`.

Reply:
Thanks for elaborating!

I saw that `String` also implements `AsRef<OsStr>` so I guess it doesn't matter as much when converting from a Rust String to an `OsString`. The other direction seems more troublesome because the `OsString` could contain non-UTF8 bytes.

Happy to stick to `OsString` here.